uaclient/snap.py

```diff
@@ -0,0 +1,54 @@
+from typing import List, Optional
+
+from uaclient import apt, util
+
+SNAP_CMD = "/usr/bin/snap"
+SNAP_INSTALL_RETRIES = [0.5, 1.0, 5.0]
+HTTP_PROXY_OPTION = "proxy.http"
+HTTPS_PROXY_OPTION = "proxy.https"
+
+
+def is_installed() -> bool:
+    """Returns whether or not snap is installed"""
+    return "snapd" in apt.get_installed_packages()
```

Comment:
@blackboxsw moving conversation to new PR (decided to create new PR because of significant differences after rebasing off of prompt PR). 

The LivepatchEntitlement does the following on enable:

```python
if not util.which(snap.SNAP_CMD):
    # install snap
elif "snapd" not in apt.get_installed_packages():
    # raise exception
```

So I wasn't sure which check I should use here. Maybe both? What do you think?